mesonbuild/programs.py

```diff
@@ -340,6 +340,12 @@ class OverrideProgram(ExternalProgram):
 
     """A script overriding a program."""
 
+    def __init__(self, *args: T.Any, version: str, **kwargs: T.Any) -> None:
```

Comment:
I know it's verbose, but I really want to stop using `Any` for things. Any instructs mypy to "ignore any errors" it's like in C writing everything as `int` and `void*` and hoping that things work out at runtime.

Reply:
At least for kwargs it is possible to pass `typing_extensions.Unpack[My_TypedDict]`. I can't remember what the state of the art might be for args, if anything.

Reply:
Indeed, what you really want is to be able to use a `ParamSpec` like you can for decorators, ie:
```python
P = typing.ParamSepc('P')

def __init__(self, *args: P.args, version: str, **kwargs: P.kwargs) -> None:
    self.version = version
    super().__init__(*args, **kwargs)
```

Of course, that doesn't work...

Reply:
The intent here is to not specify the types for `args` and `kwargs` twice, but if using T.Any disables type checking altogether, then I guess the duplication is not a problem. We can factor it out into a separate variable in the future if needed.